pkg/network/network_test.go

```diff
@@ -0,0 +1,66 @@
+	. "github.com/onsi/gomega"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+var _ = Describe("Network", func() {
+	Describe("Internal Functions", func() {
+		It("should NOT return an error because kube mac pool can be nil", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(0))
+		})
+
+		It("should NOT return an error because both or none of the KubeMacPool can be configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "", EndPoolRange: ""}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(0))
+		})
+
+		It("should return an error because only start range is configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "02:00:00:00:00:00", EndPoolRange: ""}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(1))
+			Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"))
+		})
+
+		It("should return an error because only end range is configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "", EndPoolRange: "02:00:00:FF:FF:FF"}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(1))
+			Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"))
+
+		})
+		It("should return an error because the start range mac address is invalid", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "FF:FF:FF:FF:FF:FF:FF", EndPoolRange: "02:FF:FF:FF:FF:FF"}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To((Equal(1)))
+			Expect(errorList[0].Error()).To(Equal("failed to parse startPoolRange invalid mac address"))
+		})
+
+		It("should return an error because the end range mac address is invalid", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "02:00:00:00:00:00", EndPoolRange: "FF:FF:FF:FF:FF:FF:FF"}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(1))
+			Expect(errorList[0].Error()).To(Equal("failed to parse endPoolRange invalid mac address"))
+
+		})
+
+		It("should NOT return an error mac address is valid", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "02:00:00:00:00:00", EndPoolRange: "02:FF:FF:FF:FF:FF"}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(0))
+
+		})
+
```

Comment:
please add a test with invalid MAC address

Reply:
@phoracek do you mean a MAC with the *multicast* bit on? I agree that it should be tested.

Reply:
@phoracek then we need two tests! One for multicast, one for a string that doesn't resemble a MAC address at all, "foo:bar".